src/components/Inbox/Inbox.jsx

```diff
@@ -10,16 +14,40 @@ import AppHeader from '../AppHeader';
  */
 
 const Inbox = () => {
+  const location = useLocation();
+
+  const [showForm, setShowForm] = useState(false);
+
+  localStorage.setItem('restorePath', location.pathname);
+
   return (
     <>
-      <AppHeader />
+      <Logout />
       <section id="inbox" className="panel">
-        <div>
-          Placeholder; inbox contents will go here.
-        </div>
+        <StyledButton onClick={() => (showForm ? setShowForm(false) : setShowForm(true))}>
+          New Message
+        </StyledButton>
+        {showForm && <NewMessage />}
+        <div>Placeholder; inbox contents will go here.</div>
       </section>
+      <InactivityMessage />
```

Comment:
Might need to move this to a higher level rather than inside each page. Perhaps in App.jsx or index.jsx.

Reply:
Refactored InactivityMessage to Layout. Resolving issue.